### PR TITLE
Maskbook & Twitter

### PR DESCRIPTION
**Pull request by guanbinrui**
This PR includes:

- support legacy twitter
- add two new groups for twitter #331
- fix bugs and optimize user experience

Still on going:

- update composition UI described in #397
- support dark mode && unify (#323) (blocked: #361)

### CODE REVIEW

**Comment by Jack-Works** [comment]
Should it be included in 1.8.0 release?

**Comment by yisiliu** [comment]
Let’s publish an update of this pr

**Comment by Jack-Works** [comment]
> Should it be included in 1.8.0 release?

@yisiliu @guanbinrui

**Comment by yisiliu** [comment]
> > Should it be included in 1.8.0 release?
> 
> @yisiliu @guanbinrui

> Let’s publish an update of this pr

**Comment by Jack-Works** [comment]
@guanbinrui base switched to `release/1.8.0-rc.1`

**Comment by guanbinrui** [comment]
@jackworks There are two unfixed bugs. Let me fix them before merging.

**Comment by neruthes** [comment]
Will this close #397?

**Comment by guanbinrui** [comment]
> Will this close #397?

@neruthes I will finish it in next PR.

**Comment by Jack-Works** [comment]
> Will this close #397?

I think we should not include new features in RC

**Comment by SunriseFox** [comment]
should `usexxx` be reserved for react hooks?

File: src/social-network-provider/twitter.com/ui/group.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { SocialNetworkUI } from '../../../social-network/ui'
+import { MessageCenter } from '../../../utils/messages'
+import Services from '../../../extension/service'
+import { PersonIdentifier } from '../../../database/type'
+
+export const PreDefinedTwitterGroupNames = {
```
Comment (Jack-Works) [inline src/social-network-provider/twitter.com/ui/group.ts]: Shouldn't it share with global?

Comment (guanbinrui) [reply src/social-network-provider/twitter.com/ui/group.ts]: Any better place to define it?

**Comment by Jack-Works** [comment]
merged into https://github.com/DimensionDev/Maskbook/pull/411

File: src/social-network-provider/twitter.com/ui/group.ts (hunk 1 above)
Comment (Jack-Works) [reply src/social-network-provider/twitter.com/ui/group.ts]: Maybe `PreDefinedVirtualGroupNames` in src\database\type.ts

**Pull request closed without merging**